test/common/src/main/kotlin/org/codice/compliance/verification/binding/RedirectBindingVerifier.kt

```diff
@@ -200,11 +203,16 @@ class RedirectBindingVerifier(val response: IdpRedirectResponseDecorator) {
      */
     private fun verifyRedirectDestination() {
         val destination = response.responseDom.attributes.getNamedItem("Destination")?.nodeValue
-        if (destination != ACS_URL) {
-            throw SAMLComplianceException.createWithPropertyNotEqualMessage(SAMLBindings_3_4_5_2_a1,
+        val signatures = response.responseDom.allChildren("Signature")
+
+        // "If the message is signed, the Destination XML attribute in the root SAML element of the
+        // protocol message MUST contain the URL to which the sender has instructed the user agent
+        // to deliver the message"
+        if ( signatures.isNotEmpty() && destination != TestCommon.ACS_URL) {
+            throw SAMLComplianceException.createWithPropertyNotEqualMessage(SAMLBindings_3_5_5_2_a,
                     "Destination",
                     destination,
-                    ACS_URL)
+                    TestCommon.ACS_URL)
```

Comment:
❓ Is the relevant section of spec here duplicated in both redirect and posting bindings? I'm concerned that this method is identical to the PostBinding one.

Reply:
Yes. However I feel like we shouldn't refactor until more of the verification is finished and we can see how much duplication we will have and if we want to refactor at all.